R/check.R

```diff
@@ -76,6 +76,20 @@ check_group <- function(obs) {
   return(invisible(NULL))
 }
 
+#' Add a reserved grouping variable if missing
+#'
+#' @param x A data frame or data table
```

Comment:
re my later comments - here, add_group doesn't *necessary* create a copy, but it's guaranteed to if it needs to. seems better to have it guarantee copy created (since it seems to always be used alongside a guarantee to create a copy).

Reply:
I think we want the required copy to be specified in top-level functions vs in `add_group`. For the documentation of `data.frame` I think it might be better to only ask for `data.frame` vs anything that inherits from it as anything based on a `data.frame` should be coerce-able using [data.table::as.data.table()] internally (apart from perhaps some edge cases).

Reply:
mmm, then have `add_group` require a data.table argument and expressly modify that object?